### PR TITLE
[#1835] Add `loot` data model

### PR DESCRIPTION
**Pull request by arbron**
Adds the data model for Loot items, which includes the Item Description template and the Physical Item template.

### CODE REVIEW

File: module/data/abstract.mjs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * **Note**: This uses some advanced Javascript techniques that are not necessary for most data models.
+ * Please refer to the [advancement data models]{@link BaseAdvancement} for an example of a more typical usage.
+ *
+ */
+export default class SystemDataModel extends foundry.abstract.DataModel {
+
+  /**
+   * Base templates used for construction.
+   * @type {*[]}
+   * @protected
```
Comment (Fyorl) [inline module/data/abstract.mjs:13]: We should consider `@internal` here. I don't think subclasses should really be messing with this property, it's exclusively part of the mixin mechanism.

File: module/data/item/templates/physical-item.mjs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * @property {boolean} equipped    Is this item equipped on its owning actor.
+ * @property {string} rarity       Item rarity as defined in `DND5E.itemRarity`.
+ * @property {boolean} identified  Has this item been identified?
+ * @mixin
+ */
+export default class PhysicalItemTemplate extends foundry.abstract.DataModel {
+  static defineSchema() {
+    return {
+      quantity: new foundry.data.fields.NumberField({
+        required: true, nullable: false, integer: true, positive: true, initial: 1, label: "DND5E.Quantity"
```
Comment (Fyorl) [inline module/data/item/templates/physical-item.mjs:17]: `positive: true` here is perhaps too strict. I know there are some users who like to keep things at `0` quantity so they don't forget to restock them and can do so more easily.

File: module/data/item/templates/physical-item.mjs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Migrate the item's rarity from freeform string to enum value.
+   * @param {object} source  The candidate source data from which the model will be constructed.
+   */
+  static migrateRarity(source) {
+    if ( !source.rarity ) return;
```
Comment (Fyorl) [inline module/data/item/templates/physical-item.mjs:60]:
```suggestion
    if ( !source.rarity || CONFIG.DND5E.itemRarity[key] ) return;
```
And then we can remove the or-branch further down.

**Review by Fyorl — CHANGES_REQUESTED** [review]
Nice work, Jeff. Some very minor change requests.

File: module/data/abstract.mjs (hunk 1 above)
Comment (arbron) [reply module/data/abstract.mjs:13]: `@internal` isn't a JSDoc property. We currently don't have TSDoc definitions, so that throws a linting error.

**Review by Fyorl — APPROVED** [review]
This is obviously a very well thought-out design, so thank you for putting it together, Jeff. I'm just going to append a few comments to cover all our bases, but then I think we can merge it and move onto the other data models.

**Pull request merged**